### DISCUSSION TITLE
Clarification on how to use custom containers with internal docker network and nhost environment variables

### DISCUSSION THREAD

**Discussion by caudurodev** [nhost/nhost#2465] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Hi, I would appreciate some clarity on the custom container functionality. I am trying to run a custom container via nhost that will run a script that will send and receive data from hasura graphql endpoint with admin credentials (using the hasura admin secret as credentials).  I want it to run inside the same docker network as the rest of the stack.

What will be the URL to access the graphql endpoint, assuming my container runs in the same container? The docs say hasura is in `http://hasura-service:8080` should I assume that graphql will  be`http://hasura-service:8080/v1/graphql`?

How do I share the admin-secret with my container so it can query the endpoint with credentials? Are nhost envs injected in custom containers or do we need to specify them in the environment variables for the container when spinning them up?

Some light on the subject would be appreciated!

**Comment by dbarrosop** [comment, marked as answer]
> What will be the URL to access the graphql endpoint, assuming my container runs in the same container? The docs say hasura is in http://hasura-service:8080 should I assume that graphql will behttp://hasura-service:8080/v1/graphql?

correct, paths should be the same as if you were doing `https://subdomain.hasura.region.nhost.run` (or equivalent in other services)

> How do I share the admin-secret with my container so it can query the endpoint with credentials? 

you can configure the environment variable i.e.:

<img width="688" alt="Screenshot 2024-01-12 at 10 54 52" src="https://github.com/nhost/nhost/assets/6246622/6dc67284-cfcf-4bfa-ae63-0390ab733b19">

(name doesn't matter, the important is the value)


> Are nhost envs injected in custom containers or do we need to specify them in the environment variables for the container when spinning them up?

No, to minimize the sensitive information that is exposed to containers by default we don't do that. However, you can add them yourself as needed like in the example above.

If you have more questions/comments don't hesitate to let us know.

**Comment by caudurodev** [comment]
Thanks for the clarifications! 

I noticed we can also not use `NHOST_*` environment variables but the error message is not very helpful.

**Reply by caudurodev** [reply]
Follow up questions:
- from your example above it seems we then have access to secrets - so can use those freely?
- How do I know the url of my custom container if I want to use it in a Hasura action? So hasura can send my custom container a request?

**Reply by dbarrosop** [reply]
> from your example above it seems we then have access to secrets - so can use those freely?

yes

>How do I know the url of my custom container if I want to use it in a Hasura action? So hasura can send my custom container a request?

You have that information here:

https://docs.nhost.io/guides/run/networking#connecting-to-your-service-internally
